homeassistant/components/lastfm/sensor.py

```diff
@@ -105,25 +105,24 @@ def __init__(self, user: User, entry_id: str) -> None:
     def update(self) -> None:
         """Update device state."""
         try:
-            self._user.get_playcount()
-        except WSError as exc:
+            play_count = self._user.get_playcount()
+            self._attr_entity_picture = self._user.get_image()
+            if now_playing := self._user.get_now_playing():
+                self._attr_native_value = format_track(now_playing)
+            else:
+                self._attr_native_value = STATE_NOT_SCROBBLING
+            top_played = None
+            last_played = None
+            if top_tracks := self._user.get_top_tracks(limit=1):
+                top_played = format_track(top_tracks[0].item)
+            if last_tracks := self._user.get_recent_tracks(limit=1):
+                last_played = format_track(last_tracks[0].track)
+            self._attr_extra_state_attributes = {
```

Comment:
Anything that can't raise should not be inside the try-block. 
Perhaps also valid for the format_track function?

Reply:
Do you mean I should move the format track stuff out of the try?

Reply:
Yes, should be something like this
```python
        top_played = None
        last_played = None
        try:
            play_count = self._user.get_playcount()
            self._attr_entity_picture = self._user.get_image()
            now_playing = self._user.get_now_playing()
            top_tracks = self._user.get_top_tracks(limit=1)
            last_tracks = self._user.get_recent_tracks(limit=1)
        except PyLastError as exc:
            self._attr_available = False
            LOGGER.error("Failed to load LastFM user `%s`: %r", self._user.name, exc)
            return
        self._attr_native_value = STATE_NOT_SCROBBLING
        if now_playing:
            self._attr_native_value = format_track(now_playing)
        self._attr_extra_state_attributes = {
            ATTR_LAST_PLAYED: format_track(last_tracks[0].track),
            ATTR_PLAY_COUNT: play_count,
            ATTR_TOP_PLAYED: format_track(top_tracks[0].item),
        }
```